### DISCUSSION TITLE
ObjectUtil的空判断方法

### DISCUSSION THREAD

**Discussion by amy-habor** [looly/hutool#1983] [Q&A] [answered]
public static boolean isNull(Object obj) {
	return null == obj || obj.equals(null);
}
public static boolean isNotNull(Object obj) {
	return null != obj && false == obj.equals(null);
}
上边这两段，后半部分永远都不会是true的吧，为什么还要写后半部分呢

**Comment by wangliang181230** [comment]
也许作者认为equals方法是可以重写的，有的时候在重写的equals方法里判断是null，也是有可能的。虽然当前对象不为null

**Reply by amy-habor** [reply]
这个是我没想到的，我重写equals方法第一件干的事就是入参是null返回false

**Comment by looly** [comment, marked as answer]
你可以看下JSON模块提供了`JSONNull`对象……

hutool在6.x中会去除这些让人疑惑的东西……
